src/app/workflow/permissions/permissions.component.ts

```diff
@@ -128,4 +132,20 @@ export class PermissionsComponent implements OnInit {
     this.writers = this.specificPermissionEmails(userPermissions, RoleEnum.WRITER);
     this.readers = this.specificPermissionEmails(userPermissions, RoleEnum.READER);
   }
+
+  public getWorkflowSubclass(entryType: EntryType): WorkflowSubClass {
+    switch (entryType) {
```

Comment:
Nice catch!  I've hardwired the workflow subclass to `BIOWORKFLOW`, because although it's not the best long-term solution, it's all the code needs to do for now, and what it was doing before.  In the UI Typescript, it's surprisingly hard to determine the workflow subclass (notebook, bioworkflow, apptool etc) for a given `Workflow` object.  The `type` of the workflow is returned in the JSON, but in the `openapi.yaml`, the `type` field is declared as the "discriminator" field, so maybe that's why it's not a field in the generated class?  Not sure, but we should probably modify the webservice so that the `WorkflowSubClass` is a property of `Workflow` (and its derivatives) and included with the response JSON.  Assuming everyone would be ok with that, I'll write a ticket after merging.